### PR TITLE
Fixed Examples, Added Examples CI.

### LINKED ISSUE #474: Add `examples/` into CI
Make each example case sufficiently small such that it can run on one processor for at least a few time steps. Then, add a few time steps of each example case to CI and set I/O to write at the end. Then check output for NaNs/Inftys.

We grow the size of the test suite this way, though it isn't as thorough a test as there will be no goldenfiles (though I suppose we could add them).

### PR DESCRIPTION
**Pull request by okBrian**
## Description

Reduced Example Test cases and added to CI
Fixed examples 
Fixed bugs breaking some of the examples

Currently skips these examples:
["2D_ibm_cfl_dt", "1D_sodHypo", "2D_viscous", "2D_laplace_pressure_jump", "2D_bubbly_steady_shock", "2D_advection", "2D_hardcodied_ic", "2D_ibm_multiphase"]

Fixes #474 

### To-do

- [ ] Add GH issue listing skipped example cases in CI, what it means for them to be currently "broken", and what you remember about why they are broken (NaN, tolerance, etc.). Include some possible paths to success (grid size? time step size? etc.)

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Scope

- [x] This PR comprises a set of related changes with a common goal

## How Has This Been Tested?

- [x] Github CI
- [x] Locally on a Ubuntu VM with intel compilers, and my compiler with GCC 14.2

**Test Configuration**:

* What computers and compilers did you use to test this:

## Checklist

- [x] I have added comments for the new code
- [x] I ran `./mfc.sh format` before committing my code
- [x] This PR does not introduce any repeated code (it follows the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle)
- [x] I cannot think of a way to condense this code and reduce any introduced additional line count

### CODE REVIEW

**Comment by sbryngelson** [comment]
This seems good to me. What options did you use to generate the goldenfiles? If `debug`, can you open a GitHub issue on MFC to force `debug` mode when generating new goldenfiles?

Also requesting review from @henryleberre particularly on the `toolchain/` in case there's something that should be adjusted there.

**Review by sbryngelson — APPROVED** [review]
looks good, @henryleberre may have a few comments. i put a comment on the PR for some minor details.

**Comment by okBrian** [comment]
> This seems good to me. What options did you use to generate the goldenfiles? If `debug`, can you open a GitHub issue on MFC to force `debug` mode when generating new goldenfiles?

They are generated with default settings ie (MPI, no-debug, intel: false), so no need for anything special.

File: toolchain/mfc/test/cases.py
Comment (henryleberre) [inline toolchain/mfc/test/cases.py:815]: What case does this avoid? (I might have added this..) An example directory without a case.py feels like it shouldn't exist.

File: toolchain/mfc/test/cases.py
Comment (henryleberre) [inline toolchain/mfc/test/cases.py]: you wouldn't need the comment if you had `elif caseSize > 25*25*25`.

File: toolchain/mfc/test/test.py
Comment (henryleberre) [inline toolchain/mfc/test/test.py]: I wrote another version of this logic in my latest PR which is cleaner but I'll fix the conflicts when I merge it.

File: toolchain/mfc/test/case.py
Comment (henryleberre) [inline toolchain/mfc/test/case.py]: Nitpick but the default value of `low_Mach` probably shouldn't be a boolean. If you just check equality, specifying a default value to `.get` is redundant.

File: tests/0501B3DA/golden-metadata.txt
Comment (henryleberre) [inline tests/0501B3DA/golden-metadata.txt]: I see that you regenerated a lot of existing golden files with commands like:
```
./mfc.sh test -j 8 --generate
./mfc.sh test -j 8 --generate -o 3D
```

Are all of these warranted? Given that the only fix in `src` is for analytical patches used in model patches, I don't see how this would be needed.

**Review by henryleberre — CHANGES_REQUESTED** [review]
All my comments are nitpicks except the one about regenerating too many golden files. I'll fix the nitpicks about my own code in this PR in #653 since they are fixed there (mostly).

File: tests/0501B3DA/golden-metadata.txt
Comment (sbryngelson) [reply tests/0501B3DA/golden-metadata.txt]: I'm also confused why we are re-generating goldenfiles.

Comment (okBrian) [reply tests/0501B3DA/golden-metadata.txt]: I regenerated the base test suite from when I was trying to fix the tol issues with -O0 since the errors were kind of unpredictable, so I needed to regenerate them to ensure they weren't going off of the -O0 numbers which were giving me issues.
It also probably be better to re-generate them anyways to make sure all examples and base test suite are up to date with the current code in MFC.

Comment (sbryngelson) [reply tests/0501B3DA/golden-metadata.txt]: Well we know that they are 'up to date' because they currently pass all CI, right?

Comment (henryleberre) [reply tests/0501B3DA/golden-metadata.txt]: Sure, it was useful for debugging, but are they required? As a rule of thumb, it's better not to regenerate golden files when we don't absolutely have to.

File: toolchain/mfc/test/cases.py
Comment (okBrian) [reply toolchain/mfc/test/cases.py:815]: I didn't write this portion but I removed it since an example directory without a case.py shouldn't exist.

File: toolchain/mfc/test/cases.py
Comment (okBrian) [reply toolchain/mfc/test/cases.py]: Done!

File: toolchain/mfc/test/case.py
Comment (okBrian) [reply toolchain/mfc/test/case.py]: Done!

File: tests/0501B3DA/golden-metadata.txt
Comment (okBrian) [reply tests/0501B3DA/golden-metadata.txt]: Done!

**Comment by okBrian** [comment]
Considering 175 Base Test cases are breaking on Phoenix CPU after I ran it, I am going to regenerate the base test suite.

**Comment by sbryngelson** [comment]
Why did they fail? If you didn't change how those tests were being run, the tests should pass as they do on the normal master branch, right? Or did you change the tolerances of the tests or something else?

**Comment by okBrian** [comment]
Oh yeaaa, why am I regenerating them. it seems to be an issue with mpirun, so I will look into it.

**Comment by sbryngelson** [comment]
Can be merged once CI passes + merge conflicts resolved.

**Comment by sbryngelson** [comment]
I opened a companion issue to fix the broken CI cases: https://github.com/MFlowCode/MFC/issues/751 @okBrian

**Pull request merged**
